### ISSUE TITLE
xmi-msim bottle checksums do not match

### ISSUE THREAD

**Issue by tschoonj** [Homebrew/homebrew-science#2237]
A colleague (@simionovici) reported the following error when install xmi-msim (as a dependency of xrmc):

````
/Users/alex-2454>brew install xrmc --with-xmi-msim
==> Installing xrmc from homebrew/homebrew-science
==> Installing xrmc dependency: xmi-msim
==> Downloading https://downloads.sf.net/project/machomebrew/Bottles/science/xmi-msim-5.0.yosemite.bottle.tar.gz
######################################################################## 100.0%
Error: SHA1 mismatch
Expected: f498a5402679a6ffbffb463775b04ccd7c1e94e7
Actual: 362e101038cd1a071a9db247a0f75cb11a147388
Archive: /Library/Caches/Homebrew/xmi-msim-5.0.yosemite.bottle.tar.gz
To retry an incomplete download, remove the file above.
Warning: Bottle installation failed: building from source.
Error: /usr/local/opt/autoconf not present or broken
````

I managed to reproduce this problem on my own machine and got exactly the same checksum for the bottle after downloading so it's not a connection problem or so. Could someone build the bottles again?

Also, the fallback mechanism should install the build-time dependencies before continuing with the build as shown in this case: autoconf is required for the xmi-msim build

**Comment by tdsmith** [comment]
Kicked off a build, I think: http://bot.brew.sh/job/Homebrew%20Science%20Pull%20Requests/1907/ -- will have to pull this when it's done.

I agree that build dependencies should be installed if the bottle install is abandoned; that would be a good feature request against Homebrew/homebrew if you want to raise it there.

**Comment by jacknagel** [comment]
>I agree that build dependencies should be installed if the bottle install is abandoned; that would be a good feature request against Homebrew/homebrew if you want to raise it there.

It will work for the top-level formula in a given invocation, e.g.

```
$ (unset HOMEBREW_DEVELOPER; brew install xmi-msim)
==> Installing xmi-msim from homebrew/homebrew-science
==> Downloading https://downloads.sf.net/project/machomebrew/Bottles/science/xmi-msim-5.0.yosemite.bottle.tar.gz
Already downloaded: /Library/Caches/Homebrew/xmi-msim-5.0.yosemite.bottle.tar.gz
Error: SHA1 mismatch
Expected: f498a5402679a6ffbffb463775b04ccd7c1e94e7
Actual: 362e101038cd1a071a9db247a0f75cb11a147388
Archive: /Library/Caches/Homebrew/xmi-msim-5.0.yosemite.bottle.tar.gz
To retry an incomplete download, remove the file above.
Warning: Bottle installation failed: building from source.
==> Installing xmi-msim dependency: automake
==> Downloading https://homebrew.bintray.com/bottles/automake-1.15.yosemite.bottle.tar.gz
######################################################################## 100.0%
==> Pouring automake-1.15.yosemite.bottle.tar.gz
/usr/local/Cellar/automake/1.15: 130 files, 3.2M
==> Using Homebrew-provided fortran compiler.
This may be changed by setting the FC environment variable.
==> Building with an alternative Fortran compiler
This is unsupported.
Warning: No Fortran optimization information was provided.  You may want to consider
setting FCFLAGS and FFLAGS or pass the `--default-fortran-flags` option to
`brew install` if your compiler is compatible with GCC.

If you like the default optimization level of your compiler, ignore this
warning.
==> Building with an alternative Fortran compiler
This is unsupported.
Warning: No Fortran optimization information was provided.  You may want to consider
setting FCFLAGS and FFLAGS or pass the `--default-fortran-flags` option to
`brew install` if your compiler is compatible with GCC.

If you like the default optimization level of your compiler, ignore this
warning.
==> Downloading http://lvserver.ugent.be/xmi-msim/xmimsim-5.0.tar.gz
######################################################################## 100.0%
==> Downloading https://github.com/tschoonj/xmimsim/commit/682ff5b413fc326c1cc8e9931d34bce3e920c798.diff
######################################################################## 100.0%
==> Patching
patching file configure.ac
patching file icons/Makefile.am
==> autoreconf -i
```

This is because all dependencies are computed recursively up-front and installed by the first formula installer instance, and not by the installers that get instantiated for each dependency. Fixing this is on my list for the new dependency implementation.

**Comment by tdsmith** [comment]
Thanks for the explanation, Jack!

Closed by c40f5915716fd5178858b20102b1f69bb45fa440.

**Comment by tschoonj** [comment]
@tdsmith thanks for rebuilding the bottles! I have tested them and they work fine...
